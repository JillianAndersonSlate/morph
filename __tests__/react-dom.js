const { morph } = require('../lib.js')
const { join } = require('path')
const { readdirSync, readFileSync } = require('fs')

const as = 'react-dom'
const isView = f => /\.view$/.test(f)
const getPath = (f = '.') => join(__dirname, 'views', f)

describe(as, () => {
  readdirSync(getPath()).filter(isView).forEach(f => {
    const name = f.replace(/\.view$/, '')
    const code = readFileSync(getPath(f), 'utf-8')

    it(`parses ${name}`, () => {
      expect(morph(code, { as, name, pretty: true })).toMatchSnapshot()
    })
  })
})
