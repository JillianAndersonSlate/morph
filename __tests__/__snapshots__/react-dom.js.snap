// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-dom parses Hover 1`] = `
"import React from 'react';
import ViewsAction from './ViewsAction.view';
const Hover = props => (
  <ViewsAction
    data-block-name=\\"Hover.Button\\"
    styleHover={{
      backgroundColor: 'white',
    }}
    onClick={props.onClick}
    style={{
      flexDirection: 'column',
      backgroundColor: 'black',
    }}
  >
    <div
      style={{
        color: 'white',
      }}
    >
      {'Hey!'}
    </div>
    {props.children}
  </ViewsAction>
);
export default Hover;
"
`;

exports[`react-dom parses HoverProxy 1`] = `
"import React from 'react';
import ViewsAction from './ViewsAction.view';
const HoverProxy = props => (
  <ViewsAction
    data-block-name=\\"HoverProxy.Button\\"
    styleHover={{
      backgroundColor: 'white',
    }}
    onClick={props.onClick}
    style={{
      flexDirection: 'column',
      backgroundColor: 'black',
    }}
  >
    <props.icon
      data-block-name=\\"HoverProxy.Icon\\"
      fill=\\"white\\"
      fillHover=\\"black\\"
    />
    <div
      style={{
        color: 'white',
      }}
    >
      {'Hey!'}
    </div>
    {props.children}
  </ViewsAction>
);
export default HoverProxy;
"
`;

exports[`react-dom parses JustText 1`] = `
"import React from 'react';

const JustText = props => (
  <div>
    {'hey!'}
  </div>
);
export default JustText;
"
`;

exports[`react-dom parses ListOfSomething 1`] = `
"import React from 'react';

const ListOfSomething = props => (
  <div>
    {props.list &&
      props.list.map((item, i) => (
        <div key={i}>
          {item.name}
        </div>
      ))}
  </div>
);
export default ListOfSomething;
"
`;

exports[`react-dom parses NameIsType 1`] = `
"import React from 'react';

const NameIsType = props => (
  <div data-block-name=\\"NameIsType.Texto\\">
    {'iam'}
  </div>
);
export default NameIsType;
"
`;

exports[`react-dom parses Proxy 1`] = `
"import React from 'react';

const Proxy = props => (
  <div
    data-block-name=\\"Proxy.Button\\"
    style={{
      flexDirection: 'row',
    }}
  >
    <props.icon data-block-name=\\"Proxy.Icon\\" fill=\\"red\\" />
    <div>
      {props.text}
    </div>
    {props.children}
  </div>
);
export default Proxy;
"
`;

exports[`react-dom parses UseOfProxy 1`] = `
"import React from 'react';
import Button from './Button.view';
import MyIcon from './MyIcon.view';
const UseOfProxy = props => (
  <div
    data-block-name=\\"UseOfProxy.App\\"
    style={{
      flexDirection: 'row',
    }}
  >
    <Button icon={MyIcon} text=\\"I'm a button!\\" />
    {props.children}
  </div>
);
export default UseOfProxy;
"
`;
